### ISSUE TITLE
Stackdriver: MQL group_by [interval] not supported

### ISSUE THREAD

**Issue by ChristopherWriight** [grafana/grafana#48478]
**What happened**:

The Stackdriver plugin is appending the `graph_period` to user submitted queries to control the interval of values displayed (value is controlled via the min interval UI or source). This works as long as you aren't setting your own interval in the query via other supported MQL functions. In my example, adding `group_by 1h` to the query returns the 1 hour aggregated value at the interval set by the `graph_period` function, in this case, 20 seconds. 

Example of logging metric with `group_by 1h`
<img width="1087" alt="image" src="https://user-images.githubusercontent.com/37227781/165856048-d08ebc3f-49df-44af-9f48-af7f1e65c0cd.png">

**What you expected to happen**:

The time-series interval to be controlled by the `group_by 1h` in the query.

**How to reproduce it (as minimally and precisely as possible)**:
 
- Using the stackdriver datasource
- Switch chart to Edit MQL
- Use any stackdriver metric and add, `| group_by 1h`
- Observe 1 hour aggregated values displayed at interval set by `graph_period`

**Anything else we need to know?**:

I've found that adding an additional `graph_period` to the user submitted query that matches the `group_by` interval correctly displays intended interval.

Example of logging metric with `group_by 1h | graph_period 1h`
<img width="1084" alt="image" src="https://user-images.githubusercontent.com/37227781/165856902-fc3c5ae8-7ee1-46c4-af7b-c0761dfcb19d.png">

Another concern I have with the use of `graph_period`, per documentation `graph_period` returns a min, mean, and max value when given a numeric type value, for example, so my question is - which value is being returned? 

**Environment**:
- Grafana version: v8.3.3
- Data source type & version: Stackdriver v1.0

**Comment by sarahzinger** [comment]
@ChristopherWriight  thanks for making this issue!

I spent some time investigating this today and I think if I understand this correctly the core issue here is that we always add on `graph_period` without informing users of this or giving them a way to turn it off. 

For some more repro steps on why this is a pain and how this works:

I believe the place we are adding the graph_period to queries is here: https://github.com/grafana/grafana/blob/2e9c38c95136ab4bd70ed25f67aa3fb87f5c8c4f/pkg/tsdb/cloudmonitoring/time_series_query.go#L44

To better understand the issue I commented out the line above and replaced it with:
```
timeSeriesQuery.Query += fmt.Sprintf(" | within d'%s', d'%s'", from.UTC().Format(timeFormat), to.UTC().Format(timeFormat))
```
this way I could directly edit the mql query myself without adding a graph_period with the interval by default. I also zoomed in to just a few minutes so there would be very few data points so that the discrepancies would be more obvious. 

This is the query users may think they are making:

<img width="1520" alt="Screen Shot 2022-05-02 at 8 37 32 PM" src="https://user-images.githubusercontent.com/6620164/166365729-90002116-187b-45f4-9dd3-c874b92c299b.png">

and this is the query users are actually making:
<img width="1543" alt="Screen Shot 2022-05-02 at 8 36 00 PM" src="https://user-images.githubusercontent.com/6620164/166367305-0b7eb3b5-b505-4f57-939a-67b2d453bd44.png">

The differences are subtle but if you look blue dot at the end, it is actually a slightly different value in the 2 different queries. There's nothing in our UI right now that would indicate to users that this graph_period would be tacked on by default. And as @ChristopherWriight points out adding a graph_period can return aggregate, which can mean unexpectedly different results.

In reading the code I made the assumption that a `graph_period` would be required to somehow limit the data returned to make it into something grafana could understand as [MQL's docs say](https://cloud.google.com/monitoring/mql/reference#graph_period-tabop) that this transforms the data to make it "suitable for presentation as a graph". But when I remove it, it seems fairly suitable to me?

<img width="1526" alt="Screen Shot 2022-05-02 at 8 41 47 PM" src="https://user-images.githubusercontent.com/6620164/166384154-3985da56-9353-4a0e-b961-3f6058203c0f.png">

I suspect the reason we add this by default is to ensure there are data points at expected intervals and make the graph more readable and hopefully more closely match the behavior of the datasource directly. The docs say "Metrics Explorer sets the value of the Duration argument to be appropriate for the actual time window of the chart. Explicitly giving adding graph_period as part of a query only makes sense if the query is being given to the API." But regardless, it seems odd to make this choice for users and not inform them of that, especially since it seems it can impact the behavior of other functions. For example in @ChristopherWriight's case of using `group_by` we can sometimes see very noticeably different results:

<img width="1541" alt="Screen Shot 2022-05-02 at 9 17 26 PM" src="https://user-images.githubusercontent.com/6620164/166391481-8b389271-2414-4dfe-878e-80702431d91b.png">
<img width="1542" alt="Screen Shot 2022-05-02 at 9 17 42 PM" src="https://user-images.githubusercontent.com/6620164/166391482-f921edbf-6719-4351-abb8-1788139e168a.png">

I'm going to bring this issue up with the team and see if there are opinions on how best to handle this. I can think of 3 potential options:
- remove the `graph_period` appendage entirely from the query. I suspect there are reasons why we should not and could not do this, but I'm writing it down because I want us to have very clear reasoning for why besides my hand-wavvy answer. I do suspect doing so would cause some headaches for users if they've built their queries around these limitations, so if we did it it'd have to be with care and probably saved for a major change. 
- add a toggle button to "use interval as graph_period". This would mean that all old queries could still work by having it enabled as they had been (hooray for minimizing breaking changes), but users who wanted better control over their queries could turn this functionality on/off. 
- or if adding `graph_period` by default is for some reason required in all cases (or we don't have bandwidth to address this right away), we should at the very least mention this in the ui or docs somewhere. I don't see `graph_period` mentioned anywhere in our [documentation](https://grafana.com/docs/grafana/latest/datasources/google-cloud-monitoring/).

**Comment by andresmgot** [comment]
hi @ChristopherWriight just a note to let you know that we included a fix for this in #48646 and it will be available in Grafana 9.0. Thanks for reporting the issue!
